support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller.go

```diff
@@ -161,32 +205,89 @@ func (r *PodReconciler) updateorCreatePodEntry(ctx context.Context, pod *corev1.
 }
 
 // podSpiffeID returns the desired spiffe ID for the pod, or nil if it should be ignored
-func (r *PodReconciler) podSpiffeID(pod *corev1.Pod) string {
+func (r *PodReconciler) podSpiffeID(pod *corev1.Pod) (string, error) {
 	if r.c.PodLabel != "" {
 		// the controller has been configured with a pod label. if the pod
 		// has that label, use the value to construct the pod entry. otherwise
 		// ignore the pod altogether.
 		if labelValue, ok := pod.Labels[r.c.PodLabel]; ok {
-			return makeID(r.c.TrustDomain, "%s", labelValue)
+			return makeID(r.c.TrustDomain, "%s", labelValue), nil
 		}
-		return ""
+		return "", nil
 	}
 
 	if r.c.PodAnnotation != "" {
 		// the controller has been configured with a pod annotation. if the pod
 		// has that annotation, use the value to construct the pod entry. otherwise
 		// ignore the pod altogether.
 		if annotationValue, ok := pod.Annotations[r.c.PodAnnotation]; ok {
-			return makeID(r.c.TrustDomain, "%s", annotationValue)
+			return makeID(r.c.TrustDomain, "%s", annotationValue), nil
 		}
-		return ""
+		return "", nil
 	}
 
 	// the controller has not been configured with a pod label or a pod annotation.
-	// create an entry based on the service account.
-	return makeID(r.c.TrustDomain, "ns/%s/sa/%s", pod.Namespace, pod.Spec.ServiceAccountName)
+	if r.c.IdentityTemplate != "" {
+		// create an entry using provided identity template.
+		svid, err := r.getIdentityTemplate(pod)
+		if err != nil {
+			return "", err
+		}
+
+		if r.c.IdentityTemplateLabel != "" {
```

Comment:
probably better to have this check early on so there's a quick exit without doing any extra work if there isn't a need to

Reply:
It may be nicer to check for when to return `"",nil` and then otherwise return `makeID`, reducing 2 places calling to `makeID` to just 1 place, this may make the flow a bit easier to digest.